softlayer/resource_softlayer_user.go

```diff
@@ -404,16 +429,22 @@ func resourceSoftLayerUserUpdate(d *schema.ResourceData, meta interface{}) error
 }
 
 func resourceSoftLayerUserDelete(d *schema.ResourceData, meta interface{}) error {
-	service := services.GetUserCustomerService(meta.(*session.Session))
+	sess := meta.(*session.Session)
+	service := services.GetUserCustomerService(sess)
 
 	id, _ := strconv.Atoi(d.Id())
 
+	userCancelStatusID, err := getUserStatusIDByName(sess, userCustomerCancelStatus)
```

Comment:
Is it likely for this pending cancel status ID to change? It could save roundtrips if we just have a constant for the ID.